### PR TITLE
installer: Find persist partition when installing into another disk

### PR DESCRIPTION
**Pull request by rene**
## Description

When installing persist into another storage device, the installer script must search for the persist partition in the corresponding destination and not in the main storage device.

## How to test

First, build the installer raw image (to create all needed files):

```sh
$ make installer-raw
$ make run-installer-raw # it doesn't need to run QEMU, that's only for build OVMF files
```

After that, create an additional disk image file for the persist and call qemu accordingly:

```sh
qemu-img create -f qcow2 ${EVE_REPO}/dist/amd64/current/target.img 32768M
qemu-img create -f qcow2 ${EVE_REPO}/dist/amd64/current/persist.img 32768M
qemu-system-x86_64 \
	-drive file=${EVE_REPO}/dist/amd64/current/installer.raw,format=raw \
	-drive file=${EVE_REPO}/dist/amd64/current/target.img,format=qcow2 \
	-drive file=${EVE_REPO}/dist/amd64/current/persist.img,format=qcow2 \
	-m 4096 -smp 4 -display none \
	-drive if=pflash,format=raw,unit=0,readonly,file=${EVE_REPO}/dist/amd64/current/installer/firmware/OVMF_CODE.fd \
	-drive if=pflash,format=raw,unit=1,file=${EVE_REPO}/dist/amd64/current/installer/firmware/OVMF_VARS.fd \
	-serial mon:stdio -global ICH9-LPC.noreboot=false -watchdog-action reset -rtc base=utc,clock=rt \
	-netdev user,id=eth0,net=192.168.1.0/24,dhcpstart=192.168.1.10,hostfwd=tcp::2222-:22 \
	-device virtio-net-pci,netdev=eth0,romfile="" \
	-netdev user,id=eth1,net=192.168.2.0/24,dhcpstart=192.168.2.10 -device virtio-net-pci,netdev=eth1,romfile="" \
	-device nec-usb-xhci,id=xhci \
	-qmp unix:${EVE_REPO}/qmp.sock,server,wait=off \
	-machine q35,accel=kvm,usb=off,dump-guest-core=off \
	-cpu host,invtsc=on,kvmclock=off -machine kernel-irqchip=split \
	-device intel-iommu,intremap=on,caching-mode=on,aw-bits=48 -smbios type=1,serial=31415926  
```

After boot, the GRUB options can be added manually:

```sh
eve_install_disk=sdb eve_persist_disk=sdc
```

### CODE REVIEW

File: pkg/installer/install [hunk 1]
```diff
@@ -477,7 +478,11 @@ if [ "$INSTALL_ZFS" = true ]; then
       done
   fi
 else
-  P3="$(find_part P3 "$INSTALL_DEV")"
+  if [ "$DISK_WITH_P3" != "" ]; then
+      P3="$(find_part P3 "$DISK_WITH_P3")"
+  else
+      P3="$(find_part P3 "$INSTALL_DEV")"
+  fi
   [ -z "$P3" ] && bail "Installation failed. Cannot found P3. Entering shell..."
```
Comment (deitch) [inline pkg/installer/install:486]: I am trying to understand the current and future logic.

Current logic is contained in `find_part P3 $INSTALL_DEV`, which means, "find the P3 partition on my main install disk." So if I am installing to sda, it will find P3 partition on sda.

Proposed logic is:

1. If `INSTALL_DEV != INSTALL_PERSIST`, i.e. if the 2 are separate devices, then set `DISK_WITH_P3 = $INSTALL_PERSIST` - 
2. If `DISK_WITH_P3` is not blank, then find P3 partition on `DISK_WITH_P3`, else find it on `INSTALL_DEV`

So it relies on `"$INSTALL_DEV" != "$INSTALL_PERSIST"` to determine if they are different, sets `DISK_WITH_P3` as a flag, and if the flag is set, then look on that disk. Is that it?

Makes sense, just a bit hard to read.

Would it be easier to just change line 480 to:

```sh
  P3="$(find_part P3 "$INSTALL_PERSIST")"
```

By definition, `INSTALL_PERSIST` either is equal to `INSTALL_DEV`, in which case (in PR code) `DISK_WITH_P3` is blank and therefore it is equal to the currentl or it is different, in which case we just want to look at `INSTALL_DEV`.

Comment (rene) [reply pkg/installer/install:486]: You are right @deitch , I was just trying to follow the same logic applied in the whole script (by having the DISK_WITH_P3 variable), but it doesn't make sense to turn things more complex....

**Review by eriknordmark — APPROVED** [review]
LGTM

**Review by deitch — APPROVED** [review]

**Comment by eriknordmark** [comment]
@rene does this need to be backported i.e., should we add the stable label?

**Comment by rene** [comment]
> @rene does this need to be backported i.e., should we add the stable label?

@eriknordmark , I've checked stable branches, yeah, we need to backport this fix... I'm working on it....

**Comment by rene** [comment]
Last updates:
1. Fix the mounting of persist partition (forgot to put in the PR)
2. PR was backported to stable branches

**Review by OhmSpectator — APPROVED** [review]
The change looks good.
Although the installer code is hard to follow =(

**Review by eriknordmark — APPROVED** [review]
Re-run tests

**Comment by OhmSpectator** [comment]
@rene, could you please rebase the PR to the master? It may help fix the build

**Comment by rene** [comment]
> @rene, could you please rebase the PR to the master? It may help fix the build

done.
